otx/algorithms/detection/adapters/mmdet/task.py

```diff
@@ -608,9 +626,8 @@ def hook(module, inp, outp):
             raise NotImplementedError(f"Explainer algorithm {explainer} not supported!")
         logger.info(f"Explainer algorithm: {explainer}")
 
-        # Class-wise Saliency map for Single-Stage Detector, otherwise use class-ignore saliency map.
         eval_predictions = []
-        with explainer_hook(feature_model) as saliency_hook:
+        with explainer_hook(feature_model) as saliency_hook:  # type: ignore
```

Comment:
The pylance issue with incompatible type can be solved if we'd define the type of saliency_hook above as Union[MaskRCNNRecordingForwardHook, BaseRecordingForwardHook, DetClassProbabilityMapHook].

If that's the reason why you put ignore flag here.
The same is applicable to `# type: ignore` above